docs/next/guides/building-and-testing/modules.md

```diff
@@ -481,352 +275,822 @@ new Handsontable(container, {});
+import { MergeCells } from 'handsontable/plugins/mergeCells';
+import { MultipleSelectionHandles } from 'handsontable/plugins/multipleSelectionHandles';
+import { NestedHeaders } from 'handsontable/plugins/nestedHeaders';
+import { NestedRows } from 'handsontable/plugins/nestedRows';
+import { PersistentState } from 'handsontable/plugins/persistentState';
+import { Search } from 'handsontable/plugins/search';
+import { TouchScroll } from 'handsontable/plugins/touchScroll';
+import { TrimRows } from 'handsontable/plugins/trimRows';
+import { UndoRedo } from 'handsontable/plugins/undoRedo';
+
+import { registerRenderer } from 'handsontable/renderers/registry';
+import { autocompleteRenderer } from 'handsontable/renderers/autocompleteRenderer';
+import { baseRenderer } from 'handsontable/renderers/baseRenderer';
+import { checkboxRenderer } from 'handsontable/renderers/checkboxRenderer';
+import { htmlRenderer } from 'handsontable/renderers/htmlRenderer';
+import { numericRenderer } from 'handsontable/renderers/numericRenderer';
+import { passwordRenderer } from 'handsontable/renderers/passwordRenderer';
+import { textRenderer } from 'handsontable/renderers/textRenderer';
+
+import { registerValidator } from 'handsontable/validators/registry';
+import { autocompleteValidator } from 'handsontable/validators/autocompleteValidator';
+import { dateValidator } from 'handsontable/validators/dateValidator';
+import { numericValidator } from 'handsontable/validators/numericValidator';
+import { timeValidator } from 'handsontable/validators/timeValidator';
+
+import { registerLanguageDictionary } from 'handsontable/i18n/registry';
+```
+:::
+
+## Using modules
+
+To get the most out of using Handsontable modules:
+1. [Import the required modules](#importing-required-modules).
+2. Use optional modules of your choice:
+   - [Editor modules](#using-editor-modules)
+   - [Renderer modules](#using-renderer-modules)
+   - [Validator modules](#using-validator-modules)
+   - [Cell type modules](#using-cell-type-modules)
+   - [Plugin modules](#using-plugin-modules)
+   - [Translation modules](#using-translation-modules)
+3. Remove redundant code, using [tree shaking](#tree-shaking).
+
+### Importing required modules
+
+No matter which of the [optional modules](#list-of-all-modules) you use, the following modules are always required:
+
+- Handsontable's core functionalities
+- The default [cell type](@/guides/cell-types/cell-type.md): `text`
+- [moment.js](https://momentjs.com/)
+- [DOMpurify](https://www.npmjs.com/package/dompurify)
+- [core-js](https://www.npmjs.com/package/core-js)
+- Handsontable's CSS
+
+To import the required modules, follow the steps below.
+
+#### Step 1: Remove the full Handsontable import
+
+When using modules, you no longer need to import the full version of Handsontable.
+
+Remove the `handsontable` import:
+
+```js
+// remove this line
+import Handsontable from 'handsontable';
```

Comment:
How do you now that I have this line? You could as well assume that I have blank document.

I think that it could be explained all in the next step by adding a paragraph:

> Note that to use modules, you must import `Handsontable` from **handsontable/base**, not from **handsontable** (which would give you a pre-made [package](link)).

Reply:
The assumption that the user already imported the full HoT version (and now wants to optimize it) is the same as [before](https://handsontable.com/docs/modules/#how-to-use-modules), but ok: ef4409c c56f46a

I'm not so sure about the **pre-made package** term, though. It's not used anywhere else in the docs. Can we choose one of the terms below and consistently stick to it throughout the documentation?
- The full bundle of Handsontable
- The full package of Handsontable
- The full distribution of Handsontable
- The full version of Handsontable

Reply:
Thanks, I know that you based your changed on the existing text on this page. It turns out that in my review, I addressed more than just your changes.

Regarding the "package" vocabulary, maybe you can make a new issue about that? I agree that this term needs syncing. I like the word "bundle" but I could argue that "build" would be even better. We could then call the page "Builds". I would use this term in sentences in the following way:

> The "full" and "bare" builds of Handsontable are distributed as a part of the NPM package. These builds are available for installation from NPM or for direct loading from a CDN. If you use a bundler for building your app and install Handsontable from NPM, you can also import Handsontable by specific [modules](link). Finally, it is also possible to make a [Custom build](link) of Handsontable.

Reply:
Issue created: https://github.com/handsontable/handsontable/issues/9540